File: leave-service/src/main/java/com/leave/management/leave/service/LeaveTypeService.java
package com.leave.management.leave.service;

import com.leave.management.leave.dto.LeaveTypeRequest;
import com.leave.management.leave.dto.LeaveTypeResponse;

import java.util.List;

public interface LeaveTypeService 
{
    LeaveTypeResponse createLeaveType(LeaveTypeRequest request);

    LeaveTypeResponse getLeaveTypeById(Long id);

    List<LeaveTypeResponse> getAllLeaveTypes();

    LeaveTypeResponse updateLeaveType(Long id,LeaveTypeRequest request);

    void deactivateLeaveType(Long id);
}

File: frontend/src/services/leaveTypeService.js
import axiosInstance from "./api/axiosInstance";
import API_CONFIG from "./api/apiConfig";

const leaveTypeService = {
    getAllLeaveTypes: async () => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.leaveTypes.base
        );

        return response.data;
    },

    getLeaveTypeById: async (id) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.leaveTypes.byId(id)
        );

        return response.data;
    },

    createLeaveType: async (leaveTypeData) => {
        const response = await axiosInstance.post(
            API_CONFIG.endpoints.leaveTypes.base,
            leaveTypeData
        );

        return response.data;
    },

    updateLeaveType: async (id, leaveTypeData) => {
        const response = await axiosInstance.put(
            API_CONFIG.endpoints.leaveTypes.byId(id),
            leaveTypeData
        );

        return response.data;
    },

    deactivateLeaveType: async (id) => {
        const response = await axiosInstance.patch(
            API_CONFIG.endpoints.leaveTypes.deactivate(id)
        );

        return response.data;
    },
};

export default leaveTypeService;